From a363ee8cad8e96ee1a1c3e69c6174624f9c5508b Mon Sep 17 00:00:00 2001
From: alexev <alexev@marvell.com>
Date: Mon, 17 Jun 2019 10:43:44 +0300
Subject: [PATCH 1176/1200] marvell: cn913x:  board name correction at default
 environment

Default board name of CN913x was common to
generic Armada products name in default environment.
This commit modifies the name to unique board name octeontx2_cn913x.

Change-Id: I970245948b19acc854741a27437de56b46489748
Signed-off-by: Alex Evraev <alexev@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11024
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
---
 arch/arm/mach-mvebu/Kconfig                |  9 +++
 board/Marvell/octeontx2_cn913x/MAINTAINERS |  4 ++
 board/Marvell/octeontx2_cn913x/Makefile    |  8 +++
 board/Marvell/octeontx2_cn913x/board.c     | 67 ++++++++++++++++++++++
 cmd/mvebu/Kconfig                          | 12 ++--
 configs/mvebu_db_cn91xx_defconfig          |  2 +-
 drivers/misc/mvebu_sar/Makefile            | 14 ++++-
 include/configs/octeontx2_cn913x.h         | 37 ++++++++++++
 8 files changed, 143 insertions(+), 10 deletions(-)
 create mode 100644 board/Marvell/octeontx2_cn913x/MAINTAINERS
 create mode 100644 board/Marvell/octeontx2_cn913x/Makefile
 create mode 100644 board/Marvell/octeontx2_cn913x/board.c
 create mode 100644 include/configs/octeontx2_cn913x.h

diff --git a/arch/arm/mach-mvebu/Kconfig b/arch/arm/mach-mvebu/Kconfig
index a38762db9d..1f2c19e4fa 100644
--- a/arch/arm/mach-mvebu/Kconfig
+++ b/arch/arm/mach-mvebu/Kconfig
@@ -99,6 +99,12 @@ config TARGET_MVEBU_ARMADA_8K
 	select BOARD_LATE_INIT
 	imply SCSI
 
+config TARGET_OCTEONTX2_CN913x
+	bool "Support CN913x"
+	select ARMADA_8K
+	select BOARD_LATE_INIT
+	imply SCSI
+
 config TARGET_DB_MV784MP_GP
 	bool "Support db-mv784mp-gp"
 	select MV78460
@@ -131,6 +137,7 @@ config SYS_BOARD
 	default "db-88f6820-amc" if TARGET_DB_88F6820_AMC
 	default "turris_omnia" if TARGET_TURRIS_OMNIA
 	default "mvebu_armada-8k" if TARGET_MVEBU_ARMADA_8K
+	default "octeontx2_cn913x" if TARGET_OCTEONTX2_CN913x
 	default "db-mv784mp-gp" if TARGET_DB_MV784MP_GP
 	default "ds414" if TARGET_DS414
 	default "maxbcm" if TARGET_MAXBCM
@@ -143,6 +150,7 @@ config SYS_CONFIG_NAME
 	default "db-88f6820-gp" if TARGET_DB_88F6820_GP
 	default "db-88f6820-amc" if TARGET_DB_88F6820_AMC
 	default "mvebu_armada-8k" if TARGET_MVEBU_ARMADA_8K
+	default "octeontx2_cn913x" if TARGET_OCTEONTX2_CN913x
 	default "db-mv784mp-gp" if TARGET_DB_MV784MP_GP
 	default "ds414" if TARGET_DS414
 	default "maxbcm" if TARGET_MAXBCM
@@ -156,6 +164,7 @@ config SYS_VENDOR
 	default "Marvell" if TARGET_DB_88F6820_GP
 	default "Marvell" if TARGET_DB_88F6820_AMC
 	default "Marvell" if TARGET_MVEBU_ARMADA_8K
+	default "Marvell" if TARGET_OCTEONTX2_CN913x
 	default "solidrun" if TARGET_CLEARFOG
 	default "Synology" if TARGET_DS414
 	default "CZ.NIC" if TARGET_TURRIS_OMNIA
diff --git a/board/Marvell/octeontx2_cn913x/MAINTAINERS b/board/Marvell/octeontx2_cn913x/MAINTAINERS
new file mode 100644
index 0000000000..9e390a22b1
--- /dev/null
+++ b/board/Marvell/octeontx2_cn913x/MAINTAINERS
@@ -0,0 +1,4 @@
+OCTEONTX2_CN913x BOARD
+F:	board/Marvell/octeontx2_cn913x/
+F:	include/configs/octeontx2_cn913x.h
+F:	configs/mvebu_db_cn91xx_defconfig
diff --git a/board/Marvell/octeontx2_cn913x/Makefile b/board/Marvell/octeontx2_cn913x/Makefile
new file mode 100644
index 0000000000..8c6ffb9328
--- /dev/null
+++ b/board/Marvell/octeontx2_cn913x/Makefile
@@ -0,0 +1,8 @@
+#
+# Copyright (C) 2016 Stefan Roese <sr@denx.de>
+# Copyright (C) 2019 Marvell International Ltd.
+#
+# SPDX-License-Identifier:	GPL-2.0+
+#
+
+obj-y	:= board.o
diff --git a/board/Marvell/octeontx2_cn913x/board.c b/board/Marvell/octeontx2_cn913x/board.c
new file mode 100644
index 0000000000..dced21a2ec
--- /dev/null
+++ b/board/Marvell/octeontx2_cn913x/board.c
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2016 Stefan Roese <sr@denx.de>
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#include <common.h>
+#include <dm.h>
+#include <i2c.h>
+#include <asm/io.h>
+#include <asm/arch/cpu.h>
+#include <asm/arch/soc.h>
+#include <power/regulator.h>
+#ifdef CONFIG_BOARD_CONFIG_EEPROM
+#include <mvebu/cfg_eeprom.h>
+#endif
+
+#define CP_USB20_BASE_REG(cp, p)	(MVEBU_REGS_BASE_CP(0, cp) + \
+						0x00580000 + 0x1000 * (p))
+#define CP_USB20_TX_CTRL_REG(cp, p)	(CP_USB20_BASE_REG(cp, p) + 0xC)
+#define CP_USB20_TX_OUT_AMPL_MASK	(0x7 << 20)
+#define CP_USB20_TX_OUT_AMPL_VALUE	(0x3 << 20)
+
+DECLARE_GLOBAL_DATA_PTR;
+
+int __soc_early_init_f(void)
+{
+	return 0;
+}
+
+int soc_early_init_f(void)
+			__attribute__((weak, alias("__soc_early_init_f")));
+
+int board_early_init_f(void)
+{
+	soc_early_init_f();
+
+	return 0;
+}
+
+int board_early_init_r(void)
+{
+#ifdef CONFIG_DM_REGULATOR
+	/* Check if any existing regulator should be turned down */
+	regulators_enable_boot_off(false);
+#endif
+
+	return 0;
+}
+
+int board_init(void)
+{
+	/* address of boot parameters */
+	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
+
+#ifdef CONFIG_BOARD_CONFIG_EEPROM
+	cfg_eeprom_init();
+#endif
+
+	return 0;
+}
+
+int board_late_init(void)
+{
+	return 0;
+}
diff --git a/cmd/mvebu/Kconfig b/cmd/mvebu/Kconfig
index 4524631bb7..349803cabe 100644
--- a/cmd/mvebu/Kconfig
+++ b/cmd/mvebu/Kconfig
@@ -11,7 +11,7 @@ config CMD_MVEBU_BUBT
 
 config CMD_MVEBU_MV_FW_INFO
 	bool "mv_fw_info"
-	depends on TARGET_MVEBU_ARMADA_8K
+	depends on TARGET_MVEBU_ARMADA_8K || TARGET_OCTEONTX2_CN913x
 	default y
 	help
 	  mv_fw_info command prints FreeRTOS version and firmware information
@@ -102,27 +102,27 @@ config MVEBU_UBOOT_DFLT_NAME
 
 config CMD_MVEBU_RX_TRAINING
 	bool "rx_training"
-	depends on TARGET_MVEBU_ARMADA_8K
+	depends on TARGET_MVEBU_ARMADA_8K || TARGET_OCTEONTX2_CN913x
 	default n
 	help
 	  Perform RX training sequence
 
 config CMD_MVEBU_AVS
 	bool "avs"
-	depends on TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS
+	depends on TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS || TARGET_OCTEONTX2_CN913x
 	default y
 	help
 	  Set/Get the Adaptive Voltage Scaling (AVS) value
 
 config CMD_MVEBU_MAP
 	bool "map"
-	depends on TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS
+	depends on TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS || TARGET_OCTEONTX2_CN913x
 	default y
 	help
 	  Dump address translation tables
 config CMD_MVEBU_NAND_OEM
 	bool "nand"
-	depends on (TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS || TARGET_DB_88F6820_AP) && CMD_NAND
+	depends on (TARGET_MVEBU_ARMADA_8K || TARGET_MVEBU_ARMADA_8K_PLUS || TARGET_DB_88F6820_AP || TARGET_OCTEONTX2_CN913x) && CMD_NAND
 	default y
 	help
 	  Introduction: NAND vendors conducts tests to verify if all blocks are
@@ -158,7 +158,7 @@ config CMD_MVEBU_NAND_OEM
 
 config CMD_MVEBU_MV_GET_COUNTERS
 	bool "mv_get_counters"
-	depends on TARGET_MVEBU_ARMADA_8K
+	depends on TARGET_MVEBU_ARMADA_8K || TARGET_OCTEONTX2_CN913x
 	default y
 	help
 	  mv_get_counters print MIB Counters.
diff --git a/configs/mvebu_db_cn91xx_defconfig b/configs/mvebu_db_cn91xx_defconfig
index 08ebbd08f8..119da4c81d 100644
--- a/configs/mvebu_db_cn91xx_defconfig
+++ b/configs/mvebu_db_cn91xx_defconfig
@@ -2,7 +2,7 @@ CONFIG_ARM=y
 CONFIG_ARCH_MVEBU=y
 CONFIG_SYS_TEXT_BASE=0x00000000
 CONFIG_SYS_MALLOC_F_LEN=0x2000
-CONFIG_TARGET_MVEBU_ARMADA_8K=y
+CONFIG_TARGET_OCTEONTX2_CN913x=y
 CONFIG_MVEBU_SAR=y
 CONFIG_SMBIOS_PRODUCT_NAME=""
 CONFIG_DEBUG_UART=y
diff --git a/drivers/misc/mvebu_sar/Makefile b/drivers/misc/mvebu_sar/Makefile
index 03086b88d8..412c139d1a 100644
--- a/drivers/misc/mvebu_sar/Makefile
+++ b/drivers/misc/mvebu_sar/Makefile
@@ -1,5 +1,13 @@
 
 obj-$(CONFIG_MVEBU_SAR) += sar-uclass.o
-obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap806_sar.o
-obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap807_sar.o
-obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += cp110_sar.o
+ifdef CONFIG_TARGET_MVEBU_ARMADA_8K
+	obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap806_sar.o
+	obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += ap807_sar.o
+	obj-$(CONFIG_TARGET_MVEBU_ARMADA_8K) += cp110_sar.o
+endif
+
+ifdef CONFIG_TARGET_OCTEONTX2_CN913x
+	obj-$(CONFIG_TARGET_OCTEONTX2_CN913x) += ap807_sar.o
+	obj-$(CONFIG_TARGET_OCTEONTX2_CN913x) += cp110_sar.o
+endif
+
diff --git a/include/configs/octeontx2_cn913x.h b/include/configs/octeontx2_cn913x.h
new file mode 100644
index 0000000000..a5487db725
--- /dev/null
+++ b/include/configs/octeontx2_cn913x.h
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2016 Stefan Roese <sr@denx.de>
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#ifndef _CONFIG_OCTEONTX2_CN913X_H
+#define _CONFIG_OCTEONTX2_CN913X_H
+
+#define CONFIG_DEFAULT_CONSOLE		"console=ttyS0,115200 "\
+					"earlycon=uart8250,mmio32,0xf0512000"
+
+#include <configs/mvebu_armada-common.h>
+
+/*
+ * High Level Configuration Options (easy to change)
+ */
+#define CONFIG_SYS_TCLK		250000000	/* 250MHz */
+
+#define CONFIG_NR_DRAM_BANKS	2
+
+#define CONFIG_BOARD_EARLY_INIT_R
+
+#define CONFIG_SYS_MAX_NAND_DEVICE	1
+#define CONFIG_SYS_NAND_MAX_CHIPS	1
+#define CONFIG_SYS_NAND_ONFI_DETECTION
+#define CONFIG_SYS_NAND_USE_FLASH_BBT
+
+#define CONFIG_USB_MAX_CONTROLLER_COUNT (3 + 3)
+
+/* RTC configuration */
+#ifdef CONFIG_MARVELL_RTC
+#define ERRATA_FE_3124064
+#endif
+
+#endif /* _CONFIG_OCTEONTX2_CN913X_H */
-- 
2.22.0

